### DISCUSSION TITLE
Bulk update webhooks [Oct 14]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#6774] [Discord Threads]
> 🙋‍♂️ **sevpants**

**Comment by o1lab** [comment]
> 👨‍🦲 **dstala**  
#### <@336532418100199428> If its reproducible, help us by creating an issue in github with nocodb version, root DB & other project information details captured.  
#  
> 👨‍🦲 **dstala**  
#### tried bulk update webhook on latest release & it worked for me.   
####    
#### response   
#### ```   
#### {"type":"records.after.bulkUpdate","id":"7e344850-12e1-4313-be72-9aa08b7f1651","data":{"table_id":"mutiab38fc6k5eq","table_name":"Features","previous_rows":[{"Id":1,"Title":"a"},{"Id":2,"Title":"b"},{"Id":3,"Title":"c"},{"Id":4,"Title":"d"}],"rows":[{"Id":1,"Title":"aa"},{"Id":2,"Title":"bb"},{"Id":3,"Title":"cc"},{"Id":4,"Title":"dd"}]}}   
#### ```   
####    
#### code sample   
#### ```   
#### let axios = require("axios").default;   
#### let project = 'perca85nicli62d';   
#### let table = 'mutiab38fc6k5eq';   
#### let xc_auth = 'authToken';   
####    
#### const options = {   
####     method: 'PATCH',   
####     url: `http://localhost:9000/api/v1/db/data/bulk/noco/${project}/${table}`,   
####     data: [   
####         {   
####             id: 1,   
####             Title: "aa"   
####         },   
####         {   
####             id: 2,   
####             Title: "bb"   
####         },   
####         {   
####             id: 3,   
####             Title: "cc"   
####         },   
####         {   
####             id: 4,   
####             Title: "dd"   
####         }   
####     ],   
####     headers: {   
####         'xc-auth': xc_auth   
####     }   
#### };   
####    
#### axios.request(options).then(function (response) {   
####     console.log(response.data);   
#### }).catch(function (error) {   
####     console.error(error);   
#### });   
#### ```  
#  
> 🙋‍♂️ **sevpants**  
#### I made a GitHub issue earlier today. I isolated this to being an issue with v2 triggers on smtp  
#  
> 🙋‍♂️ **sevpants**  
#### You can send a patch. That works. You can set up a url to receive the web hook. That works. Smtp does not work with BulkUpdate v2 from what I tried  
#  
> 👨‍🦲 **dstala**  
#### ok. we will check  
#  
> 🙋‍♂️ **sevpants**  
#### It's worth noting too that the sample payload in the webhooks UI is incorrect as well for the bulkupdate payload. In the sample ui, previous rows and rows are 2 separate attributes of the data attribute. Whereas with a real payload, rows are an attribute of data (not previous_rows) and then within rows there are rows and previous rows..  
#  
> 👨‍🦲 **dstala**  
#### Wasn't the case when I tried with latest release docker. The response I got is following   
####    
#### ```   
#### {"type":"records.after.bulkUpdate","id":"7e344850-12e1-4313-be72-9aa08b7f1651","data":{"table_id":"mutiab38fc6k5eq","table_name":"Features","previous_rows":[{"Id":1,"Title":"a"},{"Id":2,"Title":"b"},{"Id":3,"Title":"c"},{"Id":4,"Title":"d"}],"rows":[{"Id":1,"Title":"aa"},{"Id":2,"Title":"bb"},{"Id":3,"Title":"cc"},{"Id":4,"Title":"dd"}]}}   
#### ```  
#  
> 👨‍🦲 **dstala**  
#### Can you share more details on your project configurations?  
#  
> 🙋‍♂️ **sevpants**  
#### https://github.com/nocodb/nocodb/issues/6363  
#  
> 🙋‍♂️ **sevpants**  
#### I’m referring to the sample payload in the gui. The gui payload and the payload when you click “test webhook” return differently structured json  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1151230218992889886)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
